spv/sync.go

```diff
@@ -79,9 +79,22 @@ type Syncer struct {
 // Notifications struct to contain all of the upcoming callbacks that will
 // be used to update the rpc streams for syncing.
 type Notifications struct {
-	// Synced is defined for a callback to notify when the wallet is seen
-	// to be synced or unsynced from its connected peers.
-	Synced func(sync bool)
+	Synced                       func()
+	Unsynced                     func()
+	FetchMissingCFiltersStart    func()
+	FetchMissingCFiltersProgress func(startCFiltersHeight, endCFiltersHeight int32)
+	FetchMissingCFiltersFinished func()
+	FetchHeadersStart            func()
+	FetchHeadersProgress         func(lastHeaderHeight int32, lastHeaderTime int64)
+	FetchHeadersFinished         func()
+	DiscoverAddressesStart       func()
+	DiscoverAddressesFinished    func()
+	RescanStart                  func()
+	RescanFinished               func()
+	RescanProgress               func(rescannedThrough int32)
+
+	PeerConnected    func(peerCount int32)
+	PeerDisconnected func(peerCount int32)
```

Comment:
why are these have a newline to separate them from the rest?

Also, these will need to pass the tcp address of the peer as a param (and also add it to the protobuf message).